service/api_logic/sports_logic.py

```diff
@@ -19,22 +19,80 @@ def get_all_sports(session):
 
 
 @handle_exceptions
-def get_all_leagues_by_sport(session, dto: SportsLeagueDTO):
-    offset = (dto.page - 1) * dto.per_page
-    leagues = (
-        session.query(League)
-        .join(Sport, Sport.sport_id == League.sport_id)
-        .filter(Sport.sport_name == dto.sport)
-        .limit(dto.per_page)
-        .offset(offset)
-        .all()
+def get_all_leagues_by_sport(session, filters_dto: SportsLeagueDTO):
```

Comment:
Database-related logic belongs to DALs. No explicit invokations on database connection should be done anywhere else